task/task.go

```diff
@@ -581,6 +591,17 @@ func (m *taskManager) StopTask() {
 	}()
 }
 
+// GracefulShutdown wait all task done
+func (m *taskManager) GracefulShutdown() <-chan struct{} {
```

Comment:
How about we don't return the `chan`? I think `GracefulShutdown` means that we will block the users and then wait for the final result. So we just return nothing.

Reply:
```
   shutdown = m.GracefulShutdown()
	select {
	case <-shutdown:
		log.Printf("wait for done")
	case <-time.After(10 * time.Second):
		log.Printf("time out wait job")
	}
```
Worried about the task execution time being too long, the user can customize the timeout exit according to the situation,When all tasks are over, the channel will be closed,Don't worry about leaks.